File: Project-Euler/Q4-Largeest-Palindrome-Product/Q4.java
/*
A palindromic number reads the same both ways.
The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

Find the largest palindrome made from the product of two 3-digit numbers.
*/

public class Q4 {
  public static void main(String[] args) {
    int finalNumber = 0;

    // Loops through all pairs of 3 digit numbers
    for (int i = 100; i < 999; i++) {
      for (int j = 100; j < 999; j++) {

        int product = i * j;
        int count = 0;
        String productString = Integer.toString(product);

        for (int k = 0; k < productString.length(); k++) {
          // Compares the numbers from alternating sides (twice)
          if (productString.charAt(k) == productString.charAt(productString.length() - (1 + k))) {
            count++;
          }

          // If all sets of the numbers have been confirmed palindromes, compare it to last palindrome
          // If it's larger, replace it and print it.
          if (count == (productString.length())) {
            if (product > finalNumber) {
              finalNumber = product;
            }
          }
        }
      }
    }
    System.out.println("Largest palindrome of the product of two 3 digit numbers: " + Integer.toString(finalNumber));
  }
}

File: Project-Euler/Q4-Largeest-Palindrome-Product/Q4.py
# A palindromic number reads the same both ways.
# The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
#
# Find the largest palindrome made from the product of two 3-digit numbers.

import time
import math

start_time = time.clock()

final_number = 0

i = 100
while i < 1000:
    j = 100
    while j < 1000:
        product = i * j
        count = 0
        product_string = str(product)

        k = 0
        while k < len(product_string):
            if product_string[k] == product_string[len(product_string) - (1 + k)]:
                count += 1

            if count == len(product_string):
                if product > final_number:
                    final_number = product

            k += 1
        j += 1
    i += 1
print("Largest palindrome of the product of two 3 digit numbers:", final_number)
print("\nRuntime:", (time.clock() - start_time) * 1000, "ms")
